### PR TITLE
Search Bar Features

### PR DESCRIPTION
**Pull request by MidnightFlamingo97**
Search bar features - tested and should be fine! Also to note, my main merge from the latest main update seems to have worked perfectly! :D

### CODE REVIEW

File: scripts/script.js [hunk 1]
```diff
@@ -10,8 +13,33 @@ fetch(`cocktaildb_api_clone_local.txt`)
...
+            if (item[objKey] && !allDrinkIngredients.includes(item[objKey].toUpperCase())) {
+                allDrinkIngredients.push(item[objKey].toUpperCase());
+            }
+        }
+    });
+
+    allDrinkNamesAndIngredients = allDrinkNames.concat(allDrinkIngredients);
+
+
+    // sort array in alphabetcial order
```
Comment (thecowsays) [inline scripts/script.js:31]: Typo

**Review by thecowsays — APPROVED** [review]
Scrolled thru on my phone, looks good. I don't understand all of your code, but if it's working it should be fine.

**Review by ChrisserDev — APPROVED** [review]
All good

File: scripts/script.js (hunk 1 above)
Comment (MidnightFlamingo97) [reply scripts/script.js:31]: just in a comment, no problems

File: scripts/script.js [hunk 2]
```diff
@@ -32,4 +60,96 @@
...
+
+function search(searchString) {
+    console.log(searchInput.value.toUpperCase());
+    const isSearchingForCocktail = allDrinkNames.some(item => item === searchString);
+
+    if(isSearchingForCocktail) {
+        const cocktail = allDrinks.find(item => item.strDrink.toUpperCase() === searchString);
+        window.location.href = `./details.html?id=${cocktail.idDrink}`;
+    } else {
+        window.location.href = `./browse.html?id=${searchString}`;
```
[the remark this answers is not in the document]
Comment (thecowsays) [reply scripts/script.js:153]: image

**Pull request merged**